src/microceph.py

```diff
@@ -62,3 +73,83 @@ def add_osd_cmd(spec: str) -> None:
     """Executes MicroCeph add osd cmd with provided spec."""
     cmd = ["microceph", "disk", "add", spec]
     _run_cmd(cmd)
+
+
+def get_snap_info(snap_name):
```

Comment:
As discussed previously, I'm not fond of using HTTP requests to get this information; offline upgrades and installations should be possible, IMO. It's not required for this PR, but I think we should keep in mind that an alternative should be available in the future (i.e: An environment variable that points to a file with this information).

Reply:
We currently rely on the online availability of the snap store for installing and upgrading for downloading snaps. I'm +1 on making offline installs/upgrades possible but I'd posit that this should be part of a separate task